src/tools/wasm-interp.cc

```diff
@@ -299,8 +299,6 @@ static Result ReadAndRunModule(const char* module_filename) {
     s_stderr_stream.get()->Writef("wasi support not compiled in\n");
     return Result::Error;
 #endif
-  } else {
-    BindImports(module, imports);
   }
   BindImports(module, imports);
```

Comment:
Perhaps its this one that should be removed instead?   It looks like WASI uses `WasiBindImports` above so perhaps this  extra call is the incorrect one?     Its been a while since I looked at this code so but it seems plausible that this is why it written this way.

Reply:
Removed this one instead.